File: Baekjoon/11403_경로찾기.py
import sys
def input():
    return sys.stdin.readline().rstrip()

N = int(input())
lst = list(list(map(int,input().split())) for _ in range(N))

def dfs(idx):
    visited = [0 for _ in range(N)]
    stk = [idx]
    while stk:
        now = stk.pop()
        for x in range(N):
            if lst[now][x] == 1 and not visited[x]:
                stk.append(x)
                visited[x] = 1
                lst[idx][x] = 1

for i in range(N):
    dfs(i)

for row in lst:
    print(*row)

File: Baekjoon/js_baekjoon/11403_경로찾기.js
const fs = require('fs');
// let input = fs.readFileSync('./input.txt').toString().trim().split('\n');
let input = fs.readFileSync('./dev/stdin').toString().trim().split('\n');

const N = input[0]
const lst = input.slice(1).map((row) => row.split(" ").map(Number))

function dfs(idx) {
  const visited = Array.from({ length:N }, () => false)
  const stk = [idx]
  
  while (stk.length > 0) {
    let now = stk.pop()
    for (let x = 0; x < N; x++) {
      if (lst[now][x] == 1 && !visited[x]) {
        stk.push(x)
        visited[x] = true
        lst[idx][x] = 1
      }
    }
  }
}

for (let i = 0; i < N; i++) {
  dfs(i)
}

let result = "";
lst.forEach(function (row) {
  result += row.join(" ") + "\n"
})

console.log(result)